src/ape/types/__init__.py

```diff
@@ -226,6 +193,57 @@ def __getitem__(self, item: str) -> Any:
     def get(self, item: str, default: Optional[Any] = None) -> Any:
         return self.event_arguments.get(item, default)
 
+    class Config:
+        # NOTE: Due to https://github.com/samuelcolvin/pydantic/issues/1241 we have
+        # to add this cached property workaround in order to avoid this error:
+
+        #    TypeError: cannot pickle '_thread.RLock' object
+
+        keep_untouched = (cached_property,)
+
+
+class ContractLog(EventArguments):
+    """
+    An instance of a log from a contract.
+    """
+
+    event_name: str
+    """The name of the event."""
+
+    contract_address: AddressType
+    """The contract responsible for emitting the log."""
+
+    transaction_hash: Any
+    """The hash of the transaction containing this log."""
+
+    block_number: int
+    """The number of the block containing the transaction that produced this log."""
+
+    block_hash: Any
+    """The hash of the block containing the transaction that produced this log."""
+
+    log_index: int
+    """The index of the log on the transaction."""
+
+    transaction_index: Optional[int] = None
+    """
+    The index of the transaction's position when the log was created.
+    Is `None` when from the pending block.
+    """
+
+    @validator("block_number", "log_index", "transaction_index", pre=True)
+    def validate_hex_ints(cls, value):
+        if not isinstance(value, int):
+            return to_int(value)
+
+        return value
+
+    @validator("contract_address", pre=True)
+    def validate_address(cls, value):
+        from ape import convert
```

Comment:
Thought (not PR feedback): It might be nice to have an easier way to use Ape's conversion functionality without having to have localized imports like this.